src/utils/components/PodSelectorModal/selectorUtils.ts

```diff
@@ -0,0 +1,68 @@
+import { Selector } from '@openshift-console/dynamic-plugin-sdk';
+
+import { selectorToString } from '../Selector/utilts';
+
+import { Requirement, requirementFromString } from './selectorRequirements';
+
+export function arrayify(obj: any): string[] {
+  return Object.keys(obj).map((key) => (obj[key] ? `${key}=${obj[key]}` : key));
+}
+
+export function objectify(arr: string[]): Record<string, any> {
+  const result: Record<string, any> = {};
+  arr.forEach((item) => {
+    const [key, value = null] = item.split('=');
+    result[key] = value;
+  });
+  return result;
+}
+
+export const split = (str: string) => (str.trim() ? str.split(/,(?![^(]*\))/) : []); // [''] -> []
+
+export const cleanSelectorStr = (tag: string) => selectorToString(selectorFromString(tag));
+export const cleanTags = (currentTags: string[]) => split(cleanSelectorStr(currentTags.join(',')));
+
+export const isTagValid = (tag: string, isBasic: boolean) => {
+  const requirement = requirementFromString(tag);
+  return !!(requirement && (!isBasic || requirement.operator === 'Equals'));
+};
+
+type FromRequirementsOptions = {
+  basic?: boolean;
+  undefinedWhenEmpty?: boolean;
+};
+
+export const fromRequirements = (
+  requirements: Requirement[],
+  options: FromRequirementsOptions = {},
+): Selector => {
+  options = options || {};
+  const selector = {
```

Comment:
no need if u doing a reduce in line 49

Reply:
but if in requirements there is nothing, at least i return empty arrays

Reply:
u mean line 41? u return undefined

Reply:
when the flag `undefinedWhenEmpty` is set yes. (i took that from console. Don't know when it's used but not right know for sure)

Reply:
@upalatucci, I'm not sure I was clear enough. Maybe that will help. In line 35, you declare the selector object but only use it first on line 44. You don't need to declere it on line 35; you can simply do a requirement.reduce in line 44 and, as ACC, use a selector structure.

Reply:
clear